### PR TITLE
Split Xandra prepare and execute options

### LINKED ISSUE #47: Cannot specify Xandra query options
**Describe the bug**
Xandra query options (eg `:uuid_format`) no longer work, crashing with `** (NimbleOptions.ValidationError) unknown options [:uuid_format], valid options are: [:compressor, :force, :tracing, :custom_payload, :telemetry_metadata, :timeout]`.

I believe this is because the same options are passed to Xandra.prepare and Xandra.execute, and with https://github.com/whatyouhide/xandra/commit/f9c34bef684daae284da6d7a8ed47c689de78434 Xandra.prepare does no longer validate valid prepare options only.

**To Reproduce**
1. Perform a query with any custom option from Xandra.execute
2. The error gets thrown

**Expected behavior**
Valid options are passed to Xandra.execute

**Desktop (please complete the following information):**
 - OS: NixOS 24.05 (Uakari)
 - Elixir: 1.16.0-OTP25
 - exandra: 0.10.1
 - xandra: 0.18.1

### PR DESCRIPTION
**Pull request by noaccOS**
Fixes #47

### CODE REVIEW

File: lib/exandra.ex [hunk 1]
```diff
@@ -218,6 +218,34 @@ defmodule Exandra do
                         Xandra.Cluster
                       )
 
+  @xandra_prepare_opts [
```
Comment (whatyouhide) [inline lib/exandra.ex:221]: Instead of duplicating these lists and the logic here, can we create a `Xandra.Connection.split_prepare_and_execute_options/1` and use that here?

Comment (noaccOS) [reply lib/exandra.ex]: Done

Also fixed :telemetry_metadata and tests failing due to duplicate keys

File: lib/exandra/connection.ex [hunk 2]
```diff
@@ -967,22 +995,36 @@ defmodule Exandra.Connection do
   defp ecto_cast_to_db(:string, _query), do: "text"
   defp ecto_cast_to_db(:uuid, _query), do: "uuid"
 
+  @doc false
+  def split_prepare_and_execute_options(opts) do
+    opts =
+      Enum.uniq_by(opts, fn {key, _value} -> key end)
+      |> apply_ecto_telemetry_options_to_xandra()
```
Comment (whatyouhide) [inline lib/exandra/connection.ex:1002]:
```suggestion
    opts =
      opts
      |> Enum.uniq_by(fn {key, _value} -> key end)
      |> apply_ecto_telemetry_options_to_xandra()
```

File: lib/exandra/connection.ex [hunk 3]
```diff
@@ -967,22 +995,36 @@ defmodule Exandra.Connection do
...
+      :schema_migration,
+      :timeout,
+      :pool_size,
+      :pool,
+      :log,
+      :cast_params,
+      :prefix,
+      :cache_statement
+    ])
+    |> apply_ecto_telemetry_options_to_xandra()
```
Comment (whatyouhide) [inline lib/exandra/connection.ex:1021]:
```suggestion
    opts
    |> Keyword.drop([
      :schema_migration,
      :timeout,
      :pool_size,
      :pool,
      :log,
      :cast_params,
      :prefix,
      :cache_statement
    ])
    |> apply_ecto_telemetry_options_to_xandra()
```

File: lib/exandra.ex [hunk 4]
```diff
@@ -208,6 +208,8 @@ defmodule Exandra do
 
   use Ecto.Adapters.SQL, driver: :exandra
 
+  alias Exandra.Connection
+
```
Comment (whatyouhide) [inline lib/exandra.ex:212]:
```suggestion
```

File: lib/exandra.ex [hunk 5]
```diff
@@ -245,19 +247,8 @@ defmodule Exandra do
...
-        :tracing,
-        :compressor
-      ])
-
-    {prepare_options, execute_options} = Keyword.split(other_options, [:force])
-
-    execute_options = Keyword.merge(shared_options, execute_options)
-    prepare_options = Keyword.merge(shared_options, prepare_options)
+    {prepare_options, execute_options} =
+      Connection.split_prepare_and_execute_options(options)
```
Comment (whatyouhide) [inline lib/exandra.ex:251]:
```suggestion
      Exandra.Connection.split_prepare_and_execute_options(options)
```

File: lib/exandra.ex [hunk 6]
```diff
@@ -436,14 +427,14 @@ defmodule Exandra do
   def stream!(repo, sql, values, opts \\ []) do
     %{pid: cluster_pid} = Ecto.Repo.Registry.lookup(repo.get_dynamic_repo())
 
-    prepare_opts = Keyword.drop(opts, [:page_size])
+    {prepare_opts, execute_opts} = Connection.split_prepare_and_execute_options(opts)
```
Comment (whatyouhide) [inline lib/exandra.ex:430]:
```suggestion
    {prepare_opts, execute_opts} =
      Exandra.Connection.split_prepare_and_execute_options(opts)
```

**Review by whatyouhide — APPROVED** [review]

File: lib/exandra.ex [hunk 7]
```diff
@@ -436,14 +425,15 @@ defmodule Exandra do
   def stream!(repo, sql, values, opts \\ []) do
     %{pid: cluster_pid} = Ecto.Repo.Registry.lookup(repo.get_dynamic_repo())
 
-    prepare_opts = Keyword.drop(opts, [:page_size])
+    {prepare_opts, execute_opts} =
+      Exandra.Connection.split_prepare_and_execute_options(opts)
```
Comment (whatyouhide) [inline lib/exandra.ex:429]:
```suggestion
      Exandra.Connection.split_prepare_and_execute_options(opts)

```

**Comment by whatyouhide** [comment]
Thanks @noaccOS 💟
